federation/introspection_client.go

```diff
@@ -4,25 +4,41 @@ import (
 	"context"
 	"encoding/json"
 
+	"github.com/samsarahq/thunder/graphql"
 	"github.com/samsarahq/thunder/graphql/introspection"
 	"github.com/samsarahq/thunder/graphql/schemabuilder"
 )
 
 const IntrospectionClientName = "introspectionclient"
 
 type IntrospectionClient struct {
-	IntrospectionQueryResult []byte
+	Schema *graphql.Schema
 }
 
-func NewIntrospectionClient(res []byte) *IntrospectionClient {
+func NewIntrospectionClient(schema *graphql.Schema) *IntrospectionClient {
 	return &IntrospectionClient{
-		IntrospectionQueryResult: res,
+		Schema: schema,
 	}
 }
 
 func (c *IntrospectionClient) Execute(ctx context.Context, request *QueryRequest) (*QueryResponse, error) {
+	if err := graphql.PrepareQuery(ctx, c.Schema.Query, request.Query.SelectionSet); err != nil {
+		return nil, err
+	}
+
+	executor := graphql.NewExecutor(graphql.NewImmediateGoroutineScheduler())
```

Comment:
can we just embed the executor in the client so we don't need to create the executor each time?

Reply:
good idea